src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

```diff
@@ -264,7 +239,7 @@ public void ApplyForwarders(HttpContext context)
 
             if (checkProto)
             {
-                if (!string.IsNullOrEmpty(set.Scheme) && TryValidateScheme(set.Scheme))
+                if (!string.IsNullOrEmpty(set.Scheme) && set.Scheme.AsSpan().IndexOfAnyExcept(SchemeChars) == -1)
```

Comment:
```suggestion
                if (!string.IsNullOrEmpty(set.Scheme) && set.Scheme.AsSpan().IndexOfAnyExcept(SchemeChars) < 0)
```
Nit: `< 0` and `>= 0` checks should be slightly cheaper per @gfoidl
(applies to all such checks in the PR, but I don't know what ASP.NET's style preference is)

Reply:
Yep, as the cpu can fuse the two instructions into only one (micro-code) actually executed.